packages/keycloak-service/src/service/serviceClass.ts

```diff
@@ -105,25 +109,25 @@ export class KeycloakAPIService {
   /**
    * Constructor method
    *
-   * @param {string} accessToken - the access token
+   * @param {function() | string } accessTokenOrCallBack - asyc fn for getting the access token or access token
    * @param {string} baseURL - the base Keycloak API URL
    * @param {string} endpoint - the Keycloak endpoint
    * @param {object} getPayload - a function to get the payload
    * @param {AbortSignal} signal - signal object that allows you to communicate with a DOM request
    */
   constructor(
-    accessToken: string,
+    accessTokenOrCallBack: typeof handleSessionOrTokenExpiry = handleSessionOrTokenExpiry,
```

Comment:
Should we update the type here too to accept a string?

Reply:
addressed